src/Resource/Processor/CpuCoresCountProvider.php

```diff
@@ -65,24 +66,30 @@ public static function provide(): int
             return 1;
         }
 
-        // for Linux
-        $hasNproc = trim(@shell_exec('command -v nproc'));
+        try {
+            // for Linux
+            $hasNproc = trim(@shell_exec('command -v nproc'));
 
-        if ($hasNproc !== '') {
-            $nproc = trim(shell_exec('nproc'));
-            $cpuCount = filter_var($nproc, FILTER_VALIDATE_INT);
+            if ($hasNproc !== '') {
+                $nproc = trim(shell_exec('nproc'));
+                $cpuCount = filter_var($nproc, FILTER_VALIDATE_INT);
 
-            if (is_int($cpuCount)) {
-                return $cpuCount;
+                if (is_int($cpuCount)) {
+                    return $cpuCount;
+                }
             }
+        } catch (ExecException) {
```

Comment:
what about instead moving the existing code into a private method and catching a `Throwable` instead? From the current code I'd wager the desire behaviour is just to fallback on 1 thread instead of bubbling whatever exception happens.

Reply:
There are two separate try/catch blocks that need to be split: if one fail, the other parts need to run.

I agree with the "1 thread fallback" rule in case of error.